code/auth-service/src/permissions/permissions.js

```diff
@@ -0,0 +1,56 @@
+import fs from 'fs';
+import yaml from 'js-yaml';
+import config from '../config/config';
+
+const roleDelim = ':';
+const projectTag = 'project';
+
+const permissionAttributes = yaml.safeLoad(fs.readFileSync(config.get('permissionAttributes')));
+
+const parseRoles = role => role.split(roleDelim, 2);
+
+const concatRoles = (head, tail) =>
+  `${head}${roleDelim}${tail}`;
+
+const parseProjects = ([head, tail]) => ({
```

Comment:
This step has taken me ages to get my head around (I'm very simple). I finally realised that it is down to the fact that we have an unnecessary level in the permissions file below in the admin and project:admin split. (My fault, it is what I wrote in the story!)

I think this gets easier to code (and understand) if instead we have
instance-admin
admin
user
viewer